src/main/java/DAO/LoginDAO.java

```diff
@@ -1,14 +1,29 @@
 package DAO;
 
+import model.Login;
+import observer.LoginObserver;
+
 public class LoginDAO {
+    static LoginDAO loginDAO;
+    public Login login;
 
-    private static LoginDAO loginDAO;
-    //Todo what does this return ?
-    private String requestToken(String username, String password_hash){
-        return "";
+    public LoginDAO() {
+        login = new Login();
     }
-    //Todo
-    public static LoginDAO getInstance() {
+
+    static public LoginDAO getInstance() {
+        if (loginDAO == null) {
+            loginDAO = new LoginDAO();
+        }
         return loginDAO;
     }
+
+    public String requestToken(String username, String password_hash) {
+        login.setMessage("This is working");
+        return "Token";
+    }
+
+    public void registerObserver(LoginObserver loginObserver) {
+        login.registerObserver(loginObserver);
+    }

```

Comment:
To be honest, moet Login model niet in Controller en is DAO gwn een class die Models return?